src/components/Reports/GoalCalculator/GoalCalculator.tsx

```diff
@@ -66,41 +109,85 @@ export const GoalCalculator: React.FC<GoalCalculatorProps> = ({
     selectedCategoryId,
     selectedStepId,
     currentStep,
+    isRightOpen,
+    isDrawerOpen,
+    headerRef,
     handleCategoryChange,
     handleStepChange,
+    toggleRightPanel,
+    toggleDrawer,
+    setDrawerOpen,
+    headerHeight,
   } = useGoalCalculator();
   const { t } = useTranslation();
+  const [iconContainerWidth, setIconContainerWidth] = useState<string>('54px');
```

Comment:
Is there any way to get this to work without having to calculate the icon width? I'm probably looking at the wrong thing, but it looks fine to me without that `left: calc(${iconContainerWidth} + 1px)` prop.

Reply:
OK, so you're trying to make the right panel not cover up the icons? This is what I'm seeing with your latest commit.

<img width="860" height="844" alt="Screenshot 2025-07-31 at 4 51 55 PM" src="https://github.com/user-attachments/assets/fc5cdb84-5f85-493a-a371-30b46d4d83af" />

Another solution would be to define a constant for the icon stack width: `const iconPanelWidth = theme.spacing(4)`, use it to set the width of the stack, and use it in the `calc`. Does that make sense? That way we know the width instead of having to measure it. I'd like to avoid the measurement logic there's not another way to do this because it add complexity.

Reply:
Thanks for refactoring it to be simpler!